diff --git a/src/runtime/crt/graph_executor/graph_executor.c b/src/runtime/crt/graph_executor/graph_executor.c
index a633ed630..a31087893 100644
--- a/src/runtime/crt/graph_executor/graph_executor.c
+++ b/src/runtime/crt/graph_executor/graph_executor.c
@@ -937,7 +937,7 @@ int TVMGraphExecutor_LoadParams(TVMGraphExecutor* executor, const char* param_bl
  */
 void TVMGraphExecutor_Run(TVMGraphExecutor* executor) {
 #ifdef CONFIG_ZPL_INFERENCE_PROFILING
-  zpl_inference_enter();
+  zpl_inference_enter(0);
 #endif /* CONFIG_ZPL_INFERENCE_PROFILING */
 
   void* profiler_state;
@@ -967,7 +967,7 @@ void TVMGraphExecutor_Run(TVMGraphExecutor* executor) {
     profiler->dump_events_cb(profiler_state);
   }
 #ifdef CONFIG_ZPL_INFERENCE_PROFILING
-  zpl_inference_exit();
+  zpl_inference_exit(0);
 #endif /* CONFIG_ZPL_INFERENCE_PROFILING */
 }
 
