From 9cc0f59b233e06a5b52172c7ea58e6a11fb4fd04 Mon Sep 17 00:00:00 2001
From: Andrii Chmutov <achmutov@antmicro.com>
Date: Fri, 17 Oct 2025 15:00:19 +0200
Subject: [PATCH 09/37] check_compliance: exclude kconfig symbols

---
 scripts/ci/check_compliance.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/scripts/ci/check_compliance.py b/scripts/ci/check_compliance.py
index e2a66589204..c745cd00290 100755
--- a/scripts/ci/check_compliance.py
+++ b/scripts/ci/check_compliance.py
@@ -1498,6 +1498,11 @@ flagged.
         "WHATEVER",
         "ZEPHYR_TRY_MASS_ERASE", # MCUBoot setting described in sysbuild
                                  # documentation
+        "ZPL_SAMPLE_RUNTIME_CONF_PROFILING",
+        "ZPL_SAMPLE_TFLM_MODEL",
+        "ZPL_SAMPLE_TFLM_BUFFER_SIZE",
+        "ZPL_SAMPLE_TVM_QUANTIZED_MODEL",
+        "ZPL_SAMPLE_TVM_HEAP_SIZE",
         "ZTEST_FAIL_TEST_",  # regex in tests/ztest/fail/CMakeLists.txt
         "ZVFS_OPEN_ADD_SIZE_", # Used as an option matching prefix
         # zephyr-keep-sorted-stop
-- 
2.47.3

