From 98c27a1b212ac6da754b4db0035ecc1e0dd591aa Mon Sep 17 00:00:00 2001
From: Jedrzej Czykier <jczykier@antmicro.com>
Date: Tue, 24 Feb 2026 14:04:49 +0100
Subject: [PATCH] [#91648] boards: add sysbuild domain export

---
 boards/common/renode_robot.board.cmake | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/boards/common/renode_robot.board.cmake b/boards/common/renode_robot.board.cmake
index 3d42b3b8de1..38d345478d5 100644
--- a/boards/common/renode_robot.board.cmake
+++ b/boards/common/renode_robot.board.cmake
@@ -14,4 +14,9 @@ board_runner_args(renode-robot "--renode-robot-arg=--variable=UART:${RENODE_UART
 board_runner_args(renode-robot "--renode-robot-arg=--variable=KEYWORDS:${ZEPHYR_BASE}/tests/robot/common.robot")
 board_runner_args(renode-robot "--renode-robot-arg=--variable=RESULTS_DIR:${APPLICATION_BINARY_DIR}")

+if(SYSBUILD)
+    get_filename_component(SYSBUILD_ROOT_DIR "${CMAKE_BINARY_DIR}" DIRECTORY)
+    board_runner_args(renode-robot "--renode-robot-arg=--variable=SYSBUILD_DIR:${SYSBUILD_ROOT_DIR}")
+endif()
+
 board_finalize_runner_args(renode-robot)
--
2.47.3

