From 5f852860568372e04dab8c7abbb749a0f78d52a6 Mon Sep 17 00:00:00 2001
From: Mikolaj Klikowicz <mklikowicz@antmicro.com>
Date: Fri, 15 May 2026 11:18:26 +0200
Subject: [PATCH 39/40] scripts: ci: check_compliance: update kconfig allowlist

Signed-off-by: Mikolaj Klikowicz <mklikowicz@antmicro.com>
Signed-off-by: Grzegorz Latosinski <glatosinski@antmicro.com>
Signed-off-by: Jedrzej Czykier <jczykier@antmicro.com>
---
 scripts/ci/check_compliance.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/scripts/ci/check_compliance.py b/scripts/ci/check_compliance.py
index 5e675d9595d..dd87bc65cf2 100755
--- a/scripts/ci/check_compliance.py
+++ b/scripts/ci/check_compliance.py
@@ -1498,6 +1498,12 @@ flagged.
         "WHATEVER",
         "ZEPHYR_TRY_MASS_ERASE", # MCUBoot setting described in sysbuild
                                  # documentation
+        "ZPL_BENCHMARK_ITERS",
+        "ZPL_BENCHMARK_MAGIC",
+        "ZPL_BENCHMARK_RUN_BASIC",
+        "ZPL_BENCHMARK_RUN_DHRYSTONE",
+        "ZPL_BENCHMARK_RUN_TFLITE",
+        "ZPL_BENCHMARK_RUN_THREADS",
         "ZPL_SAMPLE_RUNTIME_CONF_PROFILING",
         "ZPL_SAMPLE_TFLM_MODEL",
         "ZPL_SAMPLE_TFLM_BUFFER_SIZE",
-- 
2.47.1

