From 4b27d2b3e5787d47b135edb733558b8cad326e13 Mon Sep 17 00:00:00 2001
From: Eryk Szpotanski <eszpotanski@antmicro.com>
Date: Fri, 31 Oct 2025 19:02:27 +0100
Subject: [PATCH] Use full thread ID

Signed-off-by: Eryk Szpotanski <eszpotanski@antmicro.com>
---
 scripts/instrumentation/zaru.py | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/scripts/instrumentation/zaru.py b/scripts/instrumentation/zaru.py
index 2251dd4840d..248aa2ecf43 100755
--- a/scripts/instrumentation/zaru.py
+++ b/scripts/instrumentation/zaru.py
@@ -768,11 +768,7 @@ def get_traces_in_trace_event_format(tmpdir, elf, demangle, verbose=False):
     named_thread_list = []
     for trace in ge:
         event_type, func, tid, cpu, _, ts, tn = trace.values()
-
-        # Use 4 LSB in tid (address) as the final thread ID just to ease
-        # displaying it in Perfetto. Hardly there will be a collision.
         tid = int(tid, 0)
-        tid = tid & 0xFFFF
 
         # Set phase type according with the Event
         if event_type == "entry":
-- 
2.51.0

