diff --git a/scripts/zaru.py b/scripts/zaru.py
index c527437d337..f491afbd843 100755
--- a/scripts/zaru.py
+++ b/scripts/zaru.py
@@ -24,6 +24,8 @@ import re
 import shutil
 import subprocess
 import tempfile
+from pathlib import Path
+from typing import Optional
 
 import serial
 from colorama import Fore, Style
@@ -44,8 +46,18 @@ ELF = "zephyr/zephyr.elf"
 
 CTF_METADATA = "ctf_metadata"
 
+# Standard directories for Zephyr SDK
+POSSIBLE_ZEPHYR_SDK_LOCATIONS = [
+    "~",
+    "~/.local",
+    "~/.local/opt",
+    "~/bin",
+    "/opt",
+    "/usr/local",
+]
 
-OBJDUMP_CMD = ["objdump", "-t"]
+OBJDUMP_CMD = None
+OBJDUMP_ARGS = ["--target=elf32-little", "-t"]
 
 
 CPPFILT_CMD = ["c++filt"]
@@ -60,6 +72,38 @@ NUM_PING_ATTEMPTS = 20
 TRACE_EVENT_FILENAME = "./trace_event.json"
 
 
+def get_zephyr_sdk() -> Optional[Path]:
+    """
+    Finds the location of Zephyr SDK.
+    """
+    full_sdk_path = os.environ.get("ZEPHYR_SDK_INSTALL_DIR", None)
+    if full_sdk_path is not None:
+        return Path(full_sdk_path)
+    zephyr_sdk = "zephyr-sdk*"
+    for possible_location in POSSIBLE_ZEPHYR_SDK_LOCATIONS:
+        path = next(
+            Path(possible_location).expanduser().glob(zephyr_sdk), None
+        )
+        if path is not None:
+            full_sdk_path = path
+            break
+    return full_sdk_path
+
+
+def get_objdump_from_sdk():
+    """
+    Finds an objdump command in Zephyr SDK.
+    """
+
+    zephyr_sdk = get_zephyr_sdk()
+    if zephyr_sdk is None:
+        return None
+
+    for objdump in zephyr_sdk.glob("*zephyr-*/bin/*zephyr-*-objdump"):
+        return objdump
+    return None
+
+
 def get_symbols_from_elf(elf_file, verbose=False):
     """Get symbols from ELF.
 
@@ -69,7 +113,14 @@ def get_symbols_from_elf(elf_file, verbose=False):
 
     assert elf_file.exists(), f"File '{elf_file}' does not exist!"
 
-    cmd = OBJDUMP_CMD + [elf_file]
+    objdump_cmd = OBJDUMP_CMD
+    if objdump_cmd is None:
+        objdump_cmd = get_objdump_from_sdk()
+    if objdump_cmd is None:
+        # Fallback to system objdump
+        objdump_cmd = "objdump"
+
+    cmd = [str(objdump_cmd)] + OBJDUMP_ARGS + [str(elf_file)]
 
     try:
         output = subprocess.check_output(cmd, text=True)
@@ -78,7 +129,7 @@ def get_symbols_from_elf(elf_file, verbose=False):
         print(f"'{cmd}' failed execution. Check if it is properly installed.")
         sys.exit(2)
     except FileNotFoundError:
-        print(f"Could not find executable '{OBJDUMP_CMD[0]}'. Please install it.")
+        print(f"Could not find executable '{objdump_cmd}'. Please install it.")
         sys.exit(3)
 
     if verbose:
